SauceExamples/testng/web/tests/InstantSauceTestNGTest4.java

```diff
@@ -0,0 +1,90 @@
+
+        /**
+         * Here we set DesiredCapabilities, in this exercise we set additional capabilities below that align with
+         * testing best practices such as timeouts, tags, and build numbers
+         */
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        capabilities.setCapability("username", sauceUserName);
+        capabilities.setCapability("accessKey", sauceAccessKey);
+        capabilities.setCapability("browserName", "Chrome");
+        capabilities.setCapability("platform", "Windows 10");
+        capabilities.setCapability("version", "59.0");
+        capabilities.setCapability("name", method.getName());
+
+        /** Tags are an excellent way to control and filter your test automation
+         * in Sauce Analytics. Get a better view into your test automation.
+         */
+        List<String> tags = Arrays.asList("sauceDemo", "demoTest", "module4");
+        capabilities.setCapability("tags", tags);
+
+        /** Another of the most important things that you can do to get started
+         * is to set timeout capabilities for Sauce based on your organizations needs. For example:
+         * How long is the whole test allowed to run?*/
+        capabilities.setCapability("maxDuration", 3600);
+        /** A Selenium crash might cause a session to hang indefinitely.
+         * Below is the max time allowed to wait for a Selenium command*/
+        capabilities.setCapability("commandTimeout", 600);
+        /** How long can the browser wait for a new command */
+        capabilities.setCapability("idleTimeout", 1000);
+
+        /** Setting a build name is one of the most fundamental pieces of running
+         * successful test automation. Builds will gather all of your tests into a single
+         * 'test suite' that you can analyze for results.
+         * It's a best practice to always group your tests into builds. */
+        capabilities.setCapability("build", "SauceDemo");
+
+        /** Don't forget to enter in your application's URL in place of 'https://www.saucedemo.com'. */
+        driver = new RemoteWebDriver(new URL("http://ondemand.saucelabs.com:80/wd/hub"), capabilities);
+        driver.navigate().to("https://www.saucedemo.com");
+
+        /** Change the value 'Swag Labs' of pageTitle to the value in your application's html page */
+        String actualPageTitle = driver.getTitle();
+        String pageTitle = "Swag Labs";
+        assertEquals(pageTitle, actualPageTitle);
+
+        /** Not sure how to grab the page title? Use a browser developer tools; for example in Chrome it's:
+         * View > Developer > Developer Tools. Then you can view the page title in the right column */
+    }
+    /**
+     * Below we are performing 2 critical actions. Quitting the driver and passing
+     * the test result to Sauce Labs user interface using the Javascript Executor.
+     */
+    @AfterMethod
+    public void cleanUpAfterTestMethod(ITestResult result) {
```

Comment:
This is great 👍 , can we apply this same idea to the JUnit tests?

Reply:
Yep, I can do that I looked up the equivalent Junit 5 class for `ITTestResult`but didn't come up with anything promising. Found `TestExecutionResults` but it looks like that pulls in a separate package than the Junit Jupiter API: https://junit.org/junit5/docs/5.0.0-M1/api/org/junit/platform/engine/TestExecutionResult.html